infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844Utils.java

```diff
@@ -88,9 +83,19 @@ public static TrustedSetup parseTrustedSetupFile(final String filePath) throws I
     }
   }
 
-  private static byte[] flattenBytes(final Stream<? extends Bytes> bytes, final int capacity) {
-    final ByteBuffer buffer = ByteBuffer.allocate(capacity);
-    bytes.map(Bytes::toArray).forEach(buffer::put);
-    return buffer.array();
+  public static byte[] flattenBytes(final List<Bytes> toFlatten, final int capacity) {
+    return flattenBytes(toFlatten, Function.identity(), capacity);
+  }
+
+  private static <T> byte[] flattenBytes(
+      final List<T> toFlatten, final Function<T, Bytes> bytesConverter, final int capacity) {
+    final byte[] flattened = new byte[capacity];
+    int destPos = 0;
+    for (final T data : toFlatten) {
+      final Bytes bytes = bytesConverter.apply(data);
+      System.arraycopy(bytes.toArray(), 0, flattened, destPos, bytes.size());
+      destPos += bytes.size();
+    }
+    return flattened;
```

Comment:
Maybe add exception here if capacity doesn't match destPos? It could lead to some weird errors if missed

Reply:
I agree, good point. Added.